src/contexts/PortalTreeContext.ts

```diff
@@ -0,0 +1,22 @@
+import { ReactPortal, createContext } from "react";
+
+/**
+ * The "Portal Tree" is a registry of React Portals, indexed by
+ * the "key" of their parent node views.
+ *
+ * Each node view registers a portal with the tree under its
+ * parent's key. Parents then retrieve the list of child portals
+ * under their key, allowing portals to be rendered with the
+ * appropriate hierarchy.
```

Comment:
Would this read better with the focus inverted?

> Each React Node View creates a unique key under which its React Node View descendants register their portals.

That might make it less necessary to put "key" in quotation marks. We're just talking about keys of a map.

We might also not want to call this a tree since this data structure itself is not one. Any other ideas?

Reply:
Yeah I'm gonna rename this "PortalRegistryContext", I think. I don't know that inverting the subject affects the readability much; it's just hard to explain this super concisely (but also this documentation is entirely internal; this isn't exported anywhere).

Reply:
> A map of portal hosts to portals.
>
> React Node Views create entries in this map under unique keys, and register themselves in the values of the keys belonging to their nearest React Node View ancestors.

How's that?

Reply:
Edited again.

Reply:
> register themselves in the value for the key belonging to their nearest React Node View ancestor.

This was challenging for me to parse

Reply:
/**
 * A map of node view keys to portals.
 *
 * Each node view registers a portal with under its parent's
 * key. Each can then retrieve the list of portals under their
 * key, allowing portals to be rendered with the appropriate
 * hierarchy.
 */

Reply:
Something like that seems fine.